tests/nsrt_learning/strips_learning/test_gen_to_spec_learner.py

```diff
@@ -326,3 +325,169 @@ def test_find_unification_and_try_specializing_pnad():
+    ], [configure_act, turn_on_act, run_act], True, 0)
+    traj2 = LowLevelTrajectory(
+        [m3_on, m1_on_m3_on, m1_on_configed, m1_on_configed_run],
+        [turn_on_act, configure_act, run_act], True, 1)
+    goal = {
+        MachineOn([m1]),
+        MachineConfigured([m1]),
+        MachineRun([m1]),
+    }
+    task1 = Task(all_off_not_configed, goal)
+    task2 = Task(m3_on, goal)
+
+    # Define the 2 demos (each with 3 segments) to backchain over.
+    segment1_1 = Segment(
+        traj1, {NotMachineOn([m1]),
+                NotMachineOn([m2]),
+                NotMachineOn([m3])}, {
+                    NotMachineOn([m1]),
+                    MachineConfigured([m1]),
+                    MachineOn([m2]),
+                    NotMachineOn([m3])
+                }, configure, goal)
+    segment1_2 = Segment(
+        traj1, {
+            NotMachineOn([m1]),
+            MachineConfigured([m1]),
+            MachineOn([m2]),
+            NotMachineOn([m3])
+        }, {
+            MachineOn([m1]),
+            MachineConfigured([m1]),
+            MachineOn([m2]),
+            NotMachineOn([m3])
+        }, turn_on, goal)
+    segment1_3 = Segment(
+        traj1, {
+            MachineOn([m1]),
+            MachineConfigured([m1]),
+            MachineOn([m2]),
+            NotMachineOn([m3])
+        }, {
+            MachineOn([m1]),
+            MachineConfigured([m1]),
+            MachineRun([m1]),
+            MachineOn([m2]),
+            NotMachineOn([m3])
+        }, run, goal)
+
+    segment2_1 = Segment(
+        traj2, {NotMachineOn([m1]),
+                NotMachineOn([m2]),
+                MachineOn([m3])},
+        {MachineOn([m1]), NotMachineOn([m2]),
+         MachineOn([m3])}, turn_on, goal)
+    segment2_2 = Segment(
+        traj2,
+        {MachineOn([m1]), NotMachineOn([m2]),
+         MachineOn([m3])}, {
+             MachineOn([m1]),
+             MachineConfigured([m1]),
+             NotMachineOn([m2]),
+             NotMachineOn([m3])
+         }, configure, goal)
+    segment2_3 = Segment(
+        traj2, {
+            MachineOn([m1]),
+            MachineConfigured([m1]),
+            NotMachineOn([m2]),
+            NotMachineOn([m3])
+        }, {
+            MachineOn([m1]),
+            MachineConfigured([m1]),
+            MachineRun([m1]),
+            NotMachineOn([m2]),
+            NotMachineOn([m3])
+        }, run, goal)
+    # Now, run the learner on the two demos and make sure check_harmlessness
+    # is set to True.
+    learner = _MockBackchainingSTRIPSLearner(
+        [traj1, traj2], [task1, task2],
+        set([MachineOn, NotMachineOn, MachineConfigured,
+             MachineRun]), [[segment1_1, segment1_2, segment1_3],
+                            [segment2_1, segment2_2, segment2_3]], True)
+    output_pnads = learner.learn()
+    # There should be exactly 3 output PNADs: 2 for Configuring, and 1 for
+    # each of TurningOn and Running.
+    assert len(output_pnads) == 4
```

Comment:
can we further verify the str() of the operators is what we would expect, like we do in the other tests? because we want to make sure e.g. that the preconditions are specific enough for both Configuring PNADs

Reply:
sounds good - i'll try to write the expected results up and include them